staging/src/k8s.io/apiserver/pkg/server/storage/storage_factory.go

```diff
@@ -252,15 +262,32 @@ func (s *DefaultStorageFactory) NewConfig(groupResource schema.GroupResource) (*
 	return &storageConfig, nil
 }
 
-// Get all backends for all registered storage destinations.
+// Backends returns all backends for all registered storage destinations.
 // Used for getting all instances for health validations.
-func (s *DefaultStorageFactory) Backends() []string {
-	backends := sets.NewString(s.StorageConfig.ServerList...)
+func (s *DefaultStorageFactory) Backends() []Backend {
+	servers := sets.NewString(s.StorageConfig.ServerList...)
 
 	for _, overrides := range s.Overrides {
-		backends.Insert(overrides.etcdLocation...)
+		servers.Insert(overrides.etcdLocation...)
+	}
+	tlsConfig := &tls.Config{
+		InsecureSkipVerify: true,
```

Comment:
would you expect to populate a RootCAs pool and unset this if StorageConfig.CAFile is set?